Sources/DDGSync/internal/RecoveryKeyTransmitter.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  RecoveryKeyTransmitter.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import os
+
+struct RecoveryKeyTransmitter: RecoveryKeyTransmitting {
+
+    let endpoints: Endpoints
+    let api: RemoteAPIRequestCreating
+    let storage: SecureStoring
+    let crypter: Crypting
+
+    func send(_ code: SyncCode.ConnectCode) async throws {
+        guard let account = try storage.account() else {
+            throw SyncError.accountNotFound
+        }
+
+        guard let token = try storage.account()?.token else {
+            throw SyncError.noToken
+        }
+
+        let recoveryKey = try JSONEncoder.snakeCaseKeys.encode(
+            SyncCode.RecoveryKey(userId: account.userId, primaryKey: account.primaryKey)
+        )
+
+        let encryptedRecoveryKey = try crypter.seal(recoveryKey, secretKey: code.secretKey)
+
+        let body = try JSONEncoder.snakeCaseKeys.encode(
+            ConnectRequest(deviceId: code.deviceId, encryptedRecoveryKey: encryptedRecoveryKey)
+        )
+
+        let request = api.createRequest(url: endpoints.connect,
+                                        method: .POST,
+                                        headers: ["Authorization": "Bearer \(token)"],
+                                        parameters: [:],
+                                        body: body,
+                                        contentType: "application/json")
```

Comment:
Tiny nitpick, this feels to me like the user of the API code has to care too much about the format of the request (like having to format the authorization header manually, and specify the content type string).

I assume we'll be needing to make similar calls in a number of other places, so it would be nice to have it handled more succinctly from the caller's perspective. What do you think?

Reply:
Agree the authorization header could be hidden away (other headers are already added elsewhere), but everything else is the responsibility of the api call to set, so laying it out like this shows the intent.  If I were to change anything anything it might be to do something like this:

`api.createAuthenticatedRequest(...)`

Maybe body could be more explicitly `jsonBody` then there'd be no need to set the content type either, I suppose.

I think we should see what we end up doing for the actual data sync and then refactor then.  I don't really want to define something now that might get changed later tbh.